db/system_keyspace.cc

```diff
@@ -1574,6 +1574,19 @@ future<> update_cdc_streams_timestamp(db_clock::time_point tp) {
             .discard_result().then([] { return force_blocking_flush(v3::CDC_LOCAL); });
 }
 
+future<> cdc_set_rewritten(std::optional<db_clock::time_point> tp) {
+    if (tp) {
+        return qctx->execute_cql(
+                format("INSERT INTO system.{} (key, streams_timestamp) VALUES (?, ?)", v3::CDC_LOCAL),
+                sstring(v3::CDC_LOCAL), *tp).discard_result();
+    } else {
+        // Insert just the row marker.
+        return qctx->execute_cql(
+                format("INSERT INTO system.{} (key) VALUES (?)", v3::CDC_LOCAL),
+                sstring(v3::CDC_LOCAL)).discard_result();
```

Comment:
Why do we have to insert anything in this case?

Reply:
Looking at `cdc_is_rewritten` it seems that not inserting anything and inserting just row marker makes no difference, no?

Reply:
```
     // We don't care about the actual timestamp; it's additional information for debugging purposes. 
```

Reply:
so you're right, we don't need the timestamp - but maybe it'll turn out to be useful

Reply:
I meant why do we need `else` here? The behaviour of `cdc_is_written` is the same no matter if we insert the marker or not.

Reply:
Ah, I see.

Not inserting anything vs inserting just the row maker **does make a difference**.
If we don't insert anything, `SELECT key FROM system.cdc_local WHERE key = rewritten` will return 0 rows
if we insert the row marker, it will return 1 row

Reply:
But for `cdc_is_written ` it makes no difference whether there's no row or 1 row with no key column -> `!msg->empty() && msg->one().has("key")`. Is there another user of this table?

Reply:
Well, first of all, `set_rewritten` has a bug, it should use the `rewritten` key, not the `cdc_local` key... So the current code would rewrite on each restart, and it would mess up the `system.cdc_local` table by setting the `cdc_local` key (which contains the current generation timestamp) to a wrong value potentially.

Now, after I fix the bug, then:
- if there's a row with `key = rewritten` containing just the row marker, then `!msg->empty() && msg->one().has("key")` will return **true** - because `key` is the partition key, so the row will show up like this:
```
cqlsh> select * from system.cdc_local ;

 key       | streams_timestamp
-----------+---------------------------------
 cdc_local | 2021-02-04 17:56:34.119000+0000
 rewritten |                            null
```
- if there is no such row, then the check will be `false`.

Reply:
There is no such thing as "row with no `key` column" because `key` is the partition key

Reply:
> There is no such thing as "row with no `key` column" because `key` is the partition key

Why do we check `msg->one().has("key")` then?

Reply:
You're right, it's not needed. But it also doesn't hurt. So I left it :P